tests/integration/utils.py

```diff
@@ -0,0 +1,221 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import os
+import yaml
+import logging
+import shutil
+import requests
+import filecmp
+from git import Repo
+from subprocess import Popen, PIPE
+from requests.packages.urllib3.util.retry import Retry
+
+
+log = logging.getLogger(__name__)
+
+class ContainerImage():
+
+    def __init__(self, repository):
+        self.repository = repository
+
+    def pull_image(self):
+        cmd = ['podman', 'pull', self.repository]
+        run_cmd(cmd, should_fail=False)
+
+        log.info('Pulled "%s"', self.repository)
+        print(f'Pulled: {self.repository}')
+
+
+    def run_cmd_on_image(self, cmd, tmpdir, should_fail):
+        image_cmd = ['podman', 'run', '-v', f'{tmpdir}:{tmpdir}', self.repository] + cmd
+        return run_cmd(image_cmd, should_fail)
+
+
+def run_cmd(cmd, should_fail):
+    log.info('Run command: %s.', cmd)
+    print(f'Run command: {cmd}')
+
+    process = Popen(cmd, stdout=PIPE, stderr=PIPE)
+    output = process.communicate()
+
+    if not should_fail and process.returncode != 0:
+        raise RuntimeError(f'Running command failed with: {output}')
+    elif should_fail and process.returncode == 0:
+        raise RuntimeError(f'Running command did not fail! Output: {output}')
+    else:
+        return (output, process.returncode)
+
+
+def clone_repository(repo, ref, tmpdir):
+    # TODO:  QUESTION - Check if repo was already cloned to tmpdir?
+    #   How? should we name folder with cloned content as "{repo}-{ref}" and if it exists
+    #       -> skip and return just absolute path to folder
```

Comment:
Cachi2 might accidentally make changes in the repository, so the content of the repo could change after running a test. For gomod specifically, Cachi2 will run various `go mod` commands which might modify files in the repo.

So I would say let's use a fresh clone every time

Reply:
ACK